dask_ml/model_selection/_incremental.py

```diff
@@ -502,21 +500,24 @@ def _get_cv_results(self, history, model_hist):
         cv_results["rank_test_score"] = order.argsort() + 1
         return cv_results
 
-    def _get_best(self, results, history_results):
-        # type: (Dict, Dict) -> Estimator
-        """Select the best estimator from the set of estimators."""
-        best_model_id = first(results.info)
-        key = operator.itemgetter("model_id")
-        best_index = -1
-        # history_results is sorted by (model_id, partial_fit_calls)
-        # best is the model_id with the highest partial fit calls
-        for k, v in itertools.groupby(history_results, key=key):
-            v = list(v)
-            best_index += len(v)
-            if k == best_model_id:
-                break
-
-        return results.models[best_model_id], best_index
```

Comment:
I'm a bit confused about the need for `_get_best` generally.  Why do we need to search for the best result throughout history.  Is this information that `_fit` could just give us directly?  I suspect that we're spending time copying how sklearn does things when it might not be the best approach overall.  Thoughts?

Reply:
>  Why do we need to search for the best result throughout history

We don't. The current implementation more or less does

``` python
score = {k: v[-1]["score"] for k, v in results.info.items()}
best_model_id = max(score, key=score.get)
```

The previous implementation picked the best model ID before searching through history.

> Is this information that _fit could just give us directly?

I refactored `_fit` to return information that's very useful in calculating that information, but not that information (the "best" model ID?) directly.

In this PR, `_fit` gives us model history separated by model ID for every model (not only the final models `_additional_calls` selects) in `results.info`. This is really useful: it allows getting the final scores for *every model* with

``` python
info, models, history = fit(...)
final_scores = {k: v[-1]["score"] for k, v in info.items()}
best_model_id = max(final_scores, key=final_scores.get)
```